pkg/telemetry/gather.go

```diff
@@ -233,3 +240,38 @@ func fetchKubernetesResources(ctx context.Context, cfg *config.Config) (map[stri
 	}
 	return metrics, nil
 }
+func getOSVersion() (string, error) {
+	file, err := os.Open("/etc/os-release")
+	if err != nil {
+		return "", fmt.Errorf("error opening /etc/os-release: %w", err)
+	}
+	defer file.Close()
+
+	var version string
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "VERSION_ID=") {
+			version = strings.TrimPrefix(line, "VERSION_ID=")
+			version = strings.Trim(version, `"`)
+			break
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return "", fmt.Errorf("error reading /etc/os-release: %w", err)
+	}
+	if version == "" {
+		return "", fmt.Errorf("VERSION_ID not found in /etc/os-release")
+	}
+	return version, nil
+}
+
+func getDeploymentType() string {
```

Comment:
Same here: should we move this to utils? Others might benefit from this function.

Reply:
This was intentionally left out of `util` because of the constants for the metrics. It is true though that they are generic enough to be in a common place. The functions called in the body are in `util` already though.